docs/guides/deployment.md

```diff
@@ -31,31 +62,127 @@ The `--bind: 8080` will run Mesop on port 8080.
 
-`requirements.txt` specifies the Python dependencies needed. You may need to add additional dependencies depending on your use case.
+In your terminal, go to the application directory, which has the files listed above.
 
-```title="requirements.txt"
-mesop
-Flask==3.0.0
-gunicorn==20.1.0
-Werkzeug==3.0.1
+Run the following command:
+
+```sh
+gcloud run deploy
 ```
 
-## Pre-requisites
+Follow the instructions and then you should be able to access your deployed app.
+
+## App Engine
 
-You will need to create a [Google Cloud](https://cloud.google.com/) account and install the [`gcloud` CLI](https://cloud.google.com/sdk/docs/install).
+### Pre-requisites
 
-## Deploy to Google Cloud Run
+You will need to create a [Google Cloud](https://cloud.google.com/) account and install the [`gcloud` CLI](https://cloud.google.com/sdk/docs/install).  See the official documentation for [detailed instructions](https://cloud.google.com/appengine/docs/flexible/python/create-app#before-you-begin).
+
+### app.yaml
+
+Create `app.yaml` to configure App Engine to run Mesop.
+
+```yaml title="app.yaml"
+runtime: python
+env: flex
+entrypoint: gunicorn -b :$PORT main:me
+
+runtime_config:
+  operating_system: ubuntu22
+  runtime_version: "3.10"
+
+manual_scaling:
+  instances: 1
+
+resources:
+  cpu: 1
+  memory_gb: 0.5
+  disk_size_gb: 10
+```
+
+## Deploy to App Engine
 
 In your terminal, go to the application directory, which has the files listed above.
 
-Run the following command:
+Run the following commands:
 
 ```sh
-$ gcloud run deploy
+gcloud app create --project=[YOUR_PROJECT_ID]
+gcloud components install app-engine-python
+gcloud app deploy
 ```
 
 Follow the instructions and then you should be able to access your deployed app.
+
+## Docker
+
+If you can run your Mesop app on Docker, you should be able to run it on many other
+cloud platforms.
+
+### Pre-requisites
+
+Make sure [Docker and Docker Compose are installed](https://docs.docker.com/engine/install/).
+
+### Dockerfile
+
+```Docker title="Dockerfile"
+FROM python:3.10.14-bullseye
+
+RUN apt-get update && \
+  apt-get install -y \
+  # General dependencies
+  locales \
+  locales-all && \
+  # Clean local repository of package files since they won't be needed anymore.
+  # Make sure this line is called after all apt-get update/install commands have
+  # run.
+  apt-get clean && \
+  # Also delete the index files which we also don't need anymore.
+  rm -rf /var/lib/apt/lists/*
+
+ENV LC_ALL en_US.UTF-8
+ENV LANG en_US.UTF-8
+ENV LANGUAGE en_US.UTF-8
+
+# Install dependencies
+COPY requirements.txt .
+RUN pip install -r requirements.txt
+
+# Create non-root user
+RUN groupadd -g 900 mesop && useradd -u 900 -s /bin/bash -g mesop mesop
+USER mesop
+
+# Add app code here
+COPY . /srv/mesop-app
+WORKDIR /srv/mesop-app
+
+# Run Mesop through gunicorn. Should be available at localhost:8080
+CMD ["gunicorn", "--bind", "0.0.0.0:8080", "main:me"]
+```
+
+### docker-compose.yaml
+
```

Comment:
I'm not super familiar with the Docker ecosystem, but I feel like it's quite common to launch docker containers directly from the CLI, perhaps include the shell command to directly run the Mesop container without docker compose? e.g.

```
docker build -t mesop-app . && docker run -p 8080:8080 mesop-app
```